enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -652,8 +652,12 @@ def can_redeem(self, request, enterprise_customer_uuid):
             )
             if not redemptions and not redeemable_policies:
                 lms_client = LmsApiClient()
-                enterprise_customer_data = lms_client.get_enterprise_customer_data(enterprise_customer_uuid)
-                enterprise_admin_users = enterprise_customer_data.get('admin_users')
+                try:
+                    enterprise_customer_data = lms_client.get_enterprise_customer_data(enterprise_customer_uuid)
+                    enterprise_admin_users = enterprise_customer_data.get('admin_users')
+                except requests.exceptions.HTTPError as exc:
+                    logger.warning(f"enterprise_customer {enterprise_customer_uuid} not found.")
+                    raise NotFound(detail="Could not find given enterprise_customer.") from exc
```

Comment:
I know it's unlikely that this call fails in isolation - is it worth failing the whole request if we can't get the admin users?  Could we try and throw a warning in the logs on exception, but continue on with the rest of the logic in this view?

Reply:
I like this idea!

Reply:
The reason I opted to fail to whole request is because this error does not pertain to any one particular requested course_key.  It is an issue of the UUID in the URL location not existing.  If we could spare an extra API call, I would have moved this check to the top of the view, but that would incur performance cost every time the view is called rather than just in error cases.

Reply:
Cool, fair point.  I think https://github.com/openedx/enterprise-access/pull/183 kind of accomplishes the "check if anything exists for the customer UUID" bit (albeit by virtue of checking for an active policy, not necessarily the existence of a customer).

Reply:
indeed, that does elegantly solve the problem.